### PR TITLE
Docs revamp

### PR DESCRIPTION
**Pull request by csadorf**

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -166,8 +177,26 @@
...
+        {'url': 'http://www.aiida.net/', 'name': 'AiiDA Home'}
+    ],
+    'github_url': 'https://github.com/aiidateam/aiida-core',
+    'use_edit_page_button': True,
+}
+html_context = {
+    'github_user': 'aiidateam',
+    'github_repo': 'aiida-core',
+    'github_version': 'master',
+    'doc_path': 'docs',
```
Comment (chrisjsewell) [inline docs/source/conf.py:192]:
```suggestion
    'doc_path': 'docs/source',
```

Comment (chrisjsewell) [reply docs/source/conf.py:192]: So that "edit this page" links to the right place

Comment (csadorf) [reply docs/source/conf.py]: Thank you!

File: docs/source/_static/aiida-custom.css [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/* Fix CSS of top bar link icons */
+a.nav-link.nav-external i {
+    padding-left: 0.3em !important;
+    font-size: inherit !important;
+    vertical-align: inherit !important;
+}
+
```
Comment (chrisjsewell) [inline docs/source/_static/aiida-custom.css:7]:
```suggestion
/* Current fix for https://github.com/pandas-dev/pydata-sphinx-theme/issues/193 */
dl.field-list {
    display: grid;
    grid-template-columns: fit-content(30%) minmax(0, 1fr);
}
```

**Comment by csadorf** [comment]
@sphuber It is not clear to me why GitHub marks the last commit as failed, but as far as I can tell, all checks passed.

File: docs/source/conf.py [hunk 3]
```diff
@@ -166,8 +177,26 @@
...
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-# TEMPORARILY DISABLED
-#html_static_path = ['_static']
+html_theme = 'pydata_sphinx_theme'
+html_theme_options = {
+    'external_links': [
+        {'url': 'http://www.aiida.net/', 'name': 'AiiDA Home'}
+    ],
+    'github_url': 'https://github.com/aiidateam/aiida-core',
```
Comment (chrisjsewell) [inline docs/source/conf.py:185]:
```suggestion
    'github_url': 'https://github.com/aiidateam/aiida-core',
    'twitter_url': 'https://twitter.com/aiidateam',
```

Comment (chrisjsewell) [reply docs/source/conf.py:185]: this is just an optional cherry on the top, we can always add it later, if you can't be bothered to re-run all the tests again

**Pull request merged**